Plutarch/DataRepr/Internal.hs

```diff
@@ -174,6 +180,52 @@ instance PPartialOrd (PDataRecord '[]) where
 
 instance POrd (PDataRecord '[])
 
+instance PShow (PDataRecord '[]) where
+  pshow' b _ = withParens b "[]"
+
+instance
+  (All Top xs, KnownSymbol label, PIsData x, PShow x, PShow (PDataRecordShowHelper xs)) =>
+  PShow (PDataRecord ((label ':= x) ': xs)) where
+  pshow' b xs = withParens b $ "[" <> pmatch xs go
+    where
+      go :: PDataRecord ((label ':= x) : xs) s -> Term s PString
+      go (PDCons y ys) =
+        mconcat
+          [ showWithLabel (Proxy @label) b y,
+            pshow' b (pcon $ PDataRecordShowHelper ys)
+          ]
+
+-- | This type exists because we need different show strategies depending
+-- on if the original list was non-empty. The idea is to implement the
+-- following at the type-level:
+--
+-- @
+-- showList' :: Show a => [a] -> String
+-- showList' [] = "[]"
+-- showList' (x:xs) = '[' : show x ++ go xs
+-- where
+--   go [] = "]"
+--   go (y:ys) = ',' : show y ++ go ys
+-- @
+newtype PDataRecordShowHelper as s = PDataRecordShowHelper (Term s (PDataRecord as))
```

Comment:
I'm honestly not a big fan of this helper datatype, there should be cleaner ways to do it, but it's not an issue honestly.
You don't have to change this.

Reply:
So normally, I'd do something with sop-core's SList/All to handle this. But it's not worth the effort IMO. It can't be made a lot cleaner.

Reply:
E.g. [`cpara_SList`](https://hackage.haskell.org/package/generics-sop-0.5.1.2/docs/Generics-SOP.html#v:cpara_SList)